typescript/CMakeLists.txt

```diff
@@ -9,3 +9,25 @@ add_custom_command(
 
 add_custom_target(generate-node_modules
                   DEPENDS "${CMAKE_BINARY_DIR}/node_modules")
+
+# Model package generation
+file(MAKE_DIRECTORY "${CMAKE_BINARY_DIR}/model.ts-package")
+
+configure_file("${CMAKE_SOURCE_DIR}/typescript/package-model.json"
+               "${CMAKE_BINARY_DIR}/model.ts-package/package.json" COPYONLY)
+configure_file("${CMAKE_SOURCE_DIR}/typescript/tsconfig-model.json"
+               "${CMAKE_BINARY_DIR}/model.ts-package/tsconfig.json" COPYONLY)
+configure_file("${CMAKE_SOURCE_DIR}/typescript/build-model.sh"
+               "${CMAKE_BINARY_DIR}/model.ts-package/build.sh" COPYONLY)
+
+add_custom_command(
+  OUTPUT "${CMAKE_BINARY_DIR}/model.ts.tgz"
+  COMMAND "${CMAKE_BINARY_DIR}/model.ts-package/build.sh"
+  WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/model.ts-package"
+  DEPENDS generate-revngModel-tuple-tree-code generate-node_modules)
+
+add_custom_target(generate-model.ts.tgz ALL
+                  DEPENDS "${CMAKE_BINARY_DIR}/model.ts.tgz")
+
+install(FILES "${CMAKE_BINARY_DIR}/model.ts.tgz"
```

Comment:
`.tar.gz`?

Reply:
Node packages have the `.tgz` extension (e.g. look in `node_cache`), so I'm sticking with that extension